zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/user/UserUpdateProcessor.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.user;
+
+import io.camunda.zeebe.engine.processing.distribution.CommandDistributionBehavior;
+import io.camunda.zeebe.engine.processing.streamprocessor.DistributedTypedRecordProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedRejectionWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedResponseWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.immutable.ProcessingState;
+import io.camunda.zeebe.engine.state.immutable.UserState;
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.UserIntent;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+import io.camunda.zeebe.stream.api.state.KeyGenerator;
+
+public class UserUpdateProcessor implements DistributedTypedRecordProcessor<UserRecord> {
+
+  private final UserState userState;
+  private final KeyGenerator keyGenerator;
+  private final StateWriter stateWriter;
+  private final TypedRejectionWriter rejectionWriter;
+  private final TypedResponseWriter responseWriter;
+  private final CommandDistributionBehavior distributionBehavior;
+
+  public UserUpdateProcessor(
+      final KeyGenerator keyGenerator,
+      final ProcessingState state,
+      final Writers writers,
+      final CommandDistributionBehavior distributionBehavior) {
+    this.keyGenerator = keyGenerator;
+    userState = state.getUserState();
+    stateWriter = writers.state();
+    rejectionWriter = writers.rejection();
+    responseWriter = writers.response();
+    this.distributionBehavior = distributionBehavior;
+  }
+
+  @Override
+  public void processNewCommand(final TypedRecord<UserRecord> command) {
+    final var username = command.getValue().getUsernameBuffer();
+    final var persistedUser = userState.getUser(username);
```

Comment:
🧠 My goal in the end would be to work with keys instead of usernames. In general keys are used as identifiers throughout Zeebe and I'd prefer to keep it consistent. This means that in the command we'd need a `userKey`, which of course gets generated by Zeebe upon user creation. The sender of this command should consume the exported `User.CREATE` record to know what the `userKey` is.

If you want we can follow up on that later in a separate PR.

Reply:
I think I'd be keen to follow up in a smaller and more focused PR for that change